src/lib.rs

```diff
@@ -198,9 +199,29 @@ impl Vessel {
     }
 }
 
+/// Guards against path strings in package data
+fn is_clean(input: &str) -> bool {
```

Comment:
Would it make sense to use something like this:
https://docs.rs/sanitize-filename/0.4.0/sanitize_filename/index.html

I'm always paranoid when I see a hand rolled sanitizer

Reply:
I tried using both [path-clean](https://docs.rs/path-clean/latest/path_clean) and [sanitize-filename](https://crates.io/crates/sanitize-filename), but they both allow various path-like filenames such as `~`, which in our case makes it possible to replace someone's home directory with a Vessel package. IMO, this makes it worth using simple, explicit [POSIX-style](https://www.gnu.org/software/automake/manual/html_node/Limitations-on-File-Names.html) validation (at least for now).